### PR TITLE
Autoreload example

### PR DESCRIPTION
**Pull request by andreyaksenov**
Added an example for the Auto-reload feature and referenced this example in a topic.

### CODE REVIEW

File: codeSnippets/snippets/autoreload/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Auto-reload
+
+A sample Ktor project showing [Auto-reload](https://ktor.io/docs/auto-reload.html) functionality. This feature works only in a development mode, which is enabled in [application.conf](resources/application.conf).
```
Comment (Stexxe) [inline codeSnippets/snippets/autoreload/README.md:3]: development mode also could be enabled by `-ea` VM option

File: codeSnippets/snippets/autoreload/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+```bash
+./gradlew -t :autoreload:build
+```
+
+Then, follow the steps below to see the Auto-reload feature in action:
+1. Open another terminal tab and run a sample:
+   ```bash
+   ./gradlew :autoreload:run
+   ```
+1. Open [http://localhost:8080/](http://localhost:8080/) to see a sample text.
```
Comment (Stexxe) [inline codeSnippets/snippets/autoreload/README.md:17]: In the browser

File: codeSnippets/snippets/autoreload/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+1. Change a text passed to the  `respondText` method in [Autoreload.kt](src/Autoreload.kt) and save the file.
```
Comment (Stexxe) [inline codeSnippets/snippets/autoreload/README.md:18]: Every point starts with `1`

File: codeSnippets/snippets/autoreload/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+## Running
+
+Since the Auto-reload feature detects changes in output files, we need to enable automatic project rebuilding. To do this, execute the following command in a repository's root directory:
+```bash
+./gradlew -t :autoreload:build
+```
+
+Then, follow the steps below to see the Auto-reload feature in action:
+1. Open another terminal tab and run a sample:
```
Comment (hhariri) [inline codeSnippets/snippets/autoreload/README.md:13]: We can drop "Then,"

File: codeSnippets/snippets/autoreload/README.md (hunk 3 above)
Comment (hhariri) [inline codeSnippets/snippets/autoreload/README.md:18]: a -> the

File: codeSnippets/snippets/autoreload/build.gradle [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+mainClassName = "io.ktor.server.netty.EngineMain"
+
+application {
+    // As an alternative way, uncomment this line to enable the development mode using the 'io.ktor.development' system property
+    // applicationDefaultJvmArgs = ["-Dio.ktor.development=true"]
+}
+
+sourceSets {
+    main.kotlin.srcDirs = [ 'src']
```
Comment (Stexxe) [inline codeSnippets/snippets/autoreload/build.gradle:22]: I don't exactly know do we agree on the standard layout for new samples or not.

File: codeSnippets/snippets/autoreload/README.md (hunk 1 above)
Comment (andreyaksenov) [reply codeSnippets/snippets/autoreload/README.md:3]: in this example, it's enabled in application.conf. Other ways are shown in a referenced topic.

File: codeSnippets/snippets/autoreload/README.md (hunk 2 above)
Comment (andreyaksenov) [reply codeSnippets/snippets/autoreload/README.md:17]: thanks, will add

**Review by hhariri — CHANGES_REQUESTED** [review]

File: codeSnippets/snippets/autoreload/build.gradle [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    repositories {
+        jcenter()
+        mavenCentral()
+    }
+    dependencies {
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+    }
+}
+
+apply plugin: 'kotlin'
```
Comment (Stexxe) [inline codeSnippets/snippets/autoreload/build.gradle:11]: This is the old way to apply plugins. Maybe we should use the new one.

File: codeSnippets/snippets/autoreload/README.md (hunk 3 above)
Comment (andreyaksenov) [reply codeSnippets/snippets/autoreload/README.md:18]: it's ok for markdown

File: codeSnippets/snippets/autoreload/README.md (hunk 1 above)
Comment (Stexxe) [reply codeSnippets/snippets/autoreload/README.md:3]: Yes, but you are describing feature here in general:
> This feature works only in a development mode, which is enabled in [application.conf](resources/application.conf)

Comment (Stexxe) [reply codeSnippets/snippets/autoreload/README.md:3]: Oh I see the `application.conf` is the concrete sample file

File: codeSnippets/snippets/autoreload/build.gradle (hunk 6 above)
Comment (andreyaksenov) [reply codeSnippets/snippets/autoreload/build.gradle]: good point, thanks. Please check a new version

File: codeSnippets/snippets/autoreload/build.gradle (hunk 5 above)
Comment (andreyaksenov) [reply codeSnippets/snippets/autoreload/build.gradle:14]: do you mean directory structure or smth else?

Comment (Stexxe) [reply codeSnippets/snippets/autoreload/build.gradle:14]: Directory structure

Comment (andreyaksenov) [reply codeSnippets/snippets/autoreload/build.gradle]: Updated a structure 
image

**Review by Stexxe — APPROVED** [review]

**Pull request merged**